Source/Renderer/FramebufferManager.js

```diff
@@ -0,0 +1,277 @@
+          sampler: Sampler.NEAREST,
+        });
+      } else {
+        this._depthStencilRenderbuffer = new Renderbuffer({
+          context: context,
+          width: width,
+          height: height,
+          format: RenderbufferFormat.DEPTH_STENCIL,
+        });
+      }
+    }
+
+    // Create depth texture
+    if (this._depth && this._createDepthAttachments) {
+      if (depthTexture) {
+        this._depthTexture = new Texture({
+          context: context,
+          width: width,
+          height: height,
+          pixelFormat: PixelFormat.DEPTH_COMPONENT,
+          pixelDatatype: PixelDatatype.UNSIGNED_INT,
+          sampler: Sampler.NEAREST,
+        });
+      } else {
+        this._depthRenderbuffer = new Renderbuffer({
+          context: context,
+          width: width,
+          height: height,
+          format: RenderbufferFormat.DEPTH_COMPONENT16,
+        });
+      }
+    }
+
+    this._framebuffer = new Framebuffer({
+      context: context,
+      colorTextures: this._colorTextures,
+      depthTexture: this._depthTexture,
+      depthRenderbuffer: this._depthRenderbuffer,
+      depthStencilTexture: this._depthStencilTexture,
+      depthStencilRenderbuffer: this._depthStencilRenderbuffer,
+      destroyAttachments: false,
+    });
+  }
+};
+
+FramebufferManager.prototype.getColorTexture = function (index) {
+  index = defaultValue(index, 0);
+  return this._colorTextures[index];
+};
+
+FramebufferManager.prototype.setColorTexture = function (texture, index) {
+  //>>includeStart('debug', pragmas.debug);
+  if (this._createColorAttachments) {
+    throw new DeveloperError(
+      "If setColorTexture is called, createColorAttachments must be false."
+    );
+  }
+  //>>includeEnd('debug');
+  index = defaultValue(index, 0);
+  this._colorTextures[index] = texture;
+};
+
+FramebufferManager.prototype.setDepthStencilTexture = function (texture) {
+  //>>includeStart('debug', pragmas.debug);
+  if (this._createDepthAttachments) {
+    throw new DeveloperError(
+      "If setDepthStencilTexture is called, createDepthAttachments must be false."
+    );
+  }
+  //>>includeEnd('debug');
+  this._depthStencilTexture = texture;
+};
+
+FramebufferManager.prototype.clear = function (
+  context,
+  passState,
+  clearColor,
+  clearCommand
+) {
+  var framebuffer = passState.framebuffer;
+  var clearCommandColor = clearCommand.color;
+
+  passState.framebuffer = this._framebuffer;
+  clearCommand.color = Color.clone(clearColor, clearCommand.color);
+  clearCommand.execute(context, passState);
+
+  clearCommand.color = clearCommandColor;
+  passState.framebuffer = framebuffer;
+};
+
+FramebufferManager.prototype.destroyResources = function () {
+  var i;
+  var length = this._colorTextures.length;
+  for (i = 0; i < length; ++i) {
+    var texture = this._colorTextures[i];
+    if (defined(texture) && !texture.isDestroyed()) {
+      this._colorTextures[i].destroy();
+    }
+  }
+  this._colorTextures = [];
+
+  if (
```

Comment:
Now that it's possible for `FramebufferManager` to store shared attachments, should it be responsible for destroying them if they belong to another class? One example that came up was `TranslucentTileClassification` having several framebuffers all using the the same `translucentDepthStencilTexture` attachment. The logic in a few other places would need to be revisited but maybe something like this would make more sense:

```js
if (!this._createXAttachments && defined(this._XTexture)) {
  this._XTexture.destroy();
}
this._XTexture = undefined;
```

For color texture attachments, all the attachments would get skipped in `destroyResources` if `createColorAttachments` was set to false.

Reply:
talked offline: owners of a `FramebufferManager` with shared attachments are responsible for deleting the shared attachments themselves. I added a `destroyFramebuffer` function to `FramebufferManager` to help in b10c8ec.